### DISCUSSION TITLE
Still a draft 0.4.0 release

### DISCUSSION THREAD

**Discussion by pvandyken** [khanlab/snakebids#90] [General]
@kaitj It looks like there's currently a draft `0.4.0` release. I remember you mentioned that was because we hadn't actually tagged our previous `0.4.0` release. Is there a plan to get this all resolved?

**Comment by akhanf** [comment]
we are a bit overdue for making that release..
also version numbers are a bit out of sync right now since the changeover -- we were at 0.4.3 before moving to the release-drafter, and now we are at 0.4.0-pre.2

Since those earlier 0.4.0 to 0.4.3 versions weren't actually tagged I am thinking it is fine to just disregard them and restart at 0.4.0 - what do you guys think? then we can just make the release asap..

**Comment by kaitj** [comment]
> I remember you mentioned that was because we hadn't actually tagged our previous 0.4.0 release

Yep, this is why the release drafter went back to `0.4.0` since it wasn't tagged. 

I'm okay with just restarting it back to `0.4.0`. I think the `0.4.0` -> `0.4.3` bump was due to upgrading `pybids`? Anyways, if we release it with what it has now, it should include everything we had wanted before! I'd vote for a `0.4.0` release.

We should also start using the labels (I'm guilty of not using them :sweat_smile:) so that the PRs get moved into the proper heading and the bumps correspond to the proper version.

**Comment by pvandyken** [comment]
Sounds good to me. Looks like `Pypi` is still on the `0.3.16` release anyway. 
> We should also start using the labels (I'm guilty of not using them 😅) so that the PRs get moved into the proper heading and the bumps correspond to the proper version.

Agreed, I've been starting to add them to my own PRs. Some of the older ones will have to be updated retroactively

**Reply by akhanf** [reply]
Yes, I just realized that all the uncategorized ones are just listed in Changes -- I just changed a label to see what happens, but is there a way to re-trigger building the release draft?

**Reply by kaitj** [reply]
Not at the moment without re-running the bump workflow (which would presumably cause another version bump). 

I could quickly set up a workflow requiring a manual trigger that runs the `release-drafter`, which should just update the draft.

**Reply by kaitj** [reply]
Trigger is implemented now to update the release-draft. Below is an example of the before and after running the workflow after adding a tag to an existing PR.

Before: 
![image](https://user-images.githubusercontent.com/21226986/145243928-01976eb5-034d-412d-b494-49e979c30a68.png)

After:
![image](https://user-images.githubusercontent.com/21226986/145244015-67a508ae-0035-4cce-9571-afbf89d940a9.png)

It triggers the same way a "deploy" should be triggered from the actions workflow tab.

**Reply by pvandyken** [reply]
Very nice. So we can retroactively go into merged PRs and add labels?

**Reply by kaitj** [reply]
Yep, though it will really only affect the PRs that have changed since the last release, so anything past that may not change (haven't tested).

**Comment by pvandyken** [comment]
Does the current deploy workflow leave room for manually editing (i.e. by typing) the release changelog. In other words, does the deploy workflow regenerate the changelog, or does it publish it as is? @kaitj

**Reply by kaitj** [reply]
The deploy workflow would regenerate the changelog since that action is being called in order to publish. You could manually edit it after the fact and it shouldn't change anything. Other option would be to update the PR prior to publishing. 

I've never really run into this so I'm not sure whether it updates to PR will update the changelog via release-drafter after it is published.

**Reply by pvandyken** [reply]
I see, okay, makes sense

**Comment by kaitj** [comment]
I've gone through and added labels to the two PRs that didn't have them for `0.4.0`. If we're happy with those, then we can run the release workflow, otherwise happy to change the labels.

@pvandyken @tkkuehn @akhanf

**Comment by akhanf** [comment]
I'm good with that! Feel free to publish the release!

**Comment by pvandyken** [comment]
Sounds good, thanks!

**Comment by akhanf** [comment]
Ok I'm publishing the release now (in case it has to be a repo admin to do it)

**Reply by akhanf** [reply]
OOPS I screwed that one up.. looks like I was supposed to run the deploy workflow instead of publishing the release..

**Reply by akhanf** [reply]
I think manually-recoverable -- I will remove the 0.4.1 pypi package, delete the v0.4.1 tag, then poetry publish the v0.4.0 tagged version..

**Reply by akhanf** [reply]
OK all set now 0.4.0 is in pypi  -- I yanked 0.4.1 from pypi, when it comes time to make that release it might need a manual deploy to force replace it, not sure, we'll see!

**Reply by pvandyken** [reply]
I can see it now!
